airbyte-to-flow/src/interceptors/fix_document_schema.rs

```diff
@@ -0,0 +1,210 @@
+
+                    prop_schema.entry("type").and_modify(|e| {
+                        if let Some(vec) = e.as_array_mut() {
+                            if let Some(null_idx) = vec.iter().position(|item| item == "null") {
+                                vec.swap_remove(null_idx);
+                            }
+                        }
+                    });
+
+                    current.push(Token::Property("properties"));
+                    current.push(Token::Property(prop));
+                },
+                doc::ptr::Token::NextIndex => todo!(),
+            }
+        }
+    }
+
+    Ok(doc)
+}
+
+#[cfg(test)]
+mod test {
+    use serde_json::{json, value::RawValue};
+
+    use super::fix_document_schema_keys;
+
+    #[test]
+    fn test_fix_document_schema_keys_prop() {
+        let doc_schema = r#"{
+            "properties": {
+                "id": {
+                    "type": ["string", "null"]
+                }
+            }
+        }"#.to_string();
+
+        let key_ptrs = vec![vec!["id".to_string()]];
+
+        assert_eq!(
+            fix_document_schema_keys(&RawValue::from_string(doc_schema).unwrap(), key_ptrs).unwrap(),
+            json!({
+                "properties": {
+                    "id": {
+                        "type": ["string"]
+                    }
+                },
+                "required": ["id"]
+            })
+        );
+    }
+
+    #[test]
+    fn test_fix_document_schema_keys_prop_deep() {
+        let doc_schema = r#"{
+            "properties": {
+                "doc": {
+                    "type": ["object", "null"],
+                    "properties": {
+                        "id": {
+                            "type": ["string", "null"]
+                        }
+                    }
+                }
+            }
+        }"#.to_string();
+
+        let key_ptrs = vec![vec!["doc".to_string(), "id".to_string()]];
+
+        assert_eq!(
+            fix_document_schema_keys(&RawValue::from_string(doc_schema).unwrap(), key_ptrs).unwrap(),
+            json!({
+                "properties": {
+                    "doc": {
+                        "type": ["object"],
+                        "properties": {
+                            "id": {
+                                "type": ["string"]
+                            }
+                        },
+                        "required": ["id"]
+                    }
+                },
+                "required": ["doc"]
+            })
+        );
+    }
+
+    #[test]
+    fn test_fix_document_schema_keys_array() {
+        let doc_schema = r#"{
+            "items": {
+                "type": ["object", "null"],
+                "properties": {
+                    "id": {
+                        "type": ["string", "null"]
+                    }
+                }
+            }
+        }"#.to_string();
+
+        let key_ptrs = vec![vec!["0".to_string(), "id".to_string()]];
```

Comment:
Related to a previous comment: This test case seems like it would be interesting in the case of a property with a name of `"0"`.